analyzers/src/SonarAnalyzer.CSharp/Rules/ClassShouldNotBeAbstract.cs

```diff
@@ -18,113 +18,79 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
-namespace SonarAnalyzer.Rules.CSharp
-{
-    [DiagnosticAnalyzer(LanguageNames.CSharp)]
-    public sealed class ClassShouldNotBeAbstract : SonarDiagnosticAnalyzer
-    {
-        private const string DiagnosticId = "S1694";
-        private const string MessageFormat = "Convert this 'abstract' {0} to {1}.";
-        private const string MessageToInterface = "an interface";
-        private const string MessageToConcreteImplementation = "a concrete type with a protected constructor";
-
-        private static readonly DiagnosticDescriptor Rule =
-            DescriptorFactory.Create(DiagnosticId, MessageFormat);
-
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
-
-        protected override void Initialize(SonarAnalysisContext context) =>
-            context.RegisterSymbolAction(
-                c =>
-                {
-                    var symbol = (INamedTypeSymbol)c.Symbol;
-                    if (!symbol.IsClass()
-                        || !symbol.IsAbstract
-                        || HasInheritedAbstractMembers(symbol))
-                    {
-                        return;
-                    }
-
-                    if (!IsRecordWithParameters(symbol) && AbstractTypeShouldBeInterface(symbol))
-                    {
-                        Report(c, symbol, MessageToInterface);
-                        return;
-                    }
-
-                    if (AbstractTypeShouldBeConcrete(symbol))
-                    {
-                        Report(c, symbol, MessageToConcreteImplementation);
-                    }
-                },
-                SymbolKind.NamedType);
+namespace SonarAnalyzer.Rules.CSharp;
 
-        private static bool HasInheritedAbstractMembers(INamedTypeSymbol symbol)
-        {
-            var baseTypes = symbol.BaseType.GetSelfAndBaseTypes().ToList();
-            var abstractMethods = baseTypes.SelectMany(GetAllAbstractMethods);
-            var baseTypesAndSelf = baseTypes.Concat(new[] { symbol }).ToList();
-            var overrideMethods = baseTypesAndSelf.SelectMany(GetAllOverrideMethods);
-            var overriddenMethods = overrideMethods.Select(m => m.OverriddenMethod);
-            var stillAbstractMethods = abstractMethods.Except(overriddenMethods);
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public sealed class ClassShouldNotBeAbstract : SonarDiagnosticAnalyzer<SyntaxKind>
```

Comment:
The name of this class is not the best IMO, what about `AbstractClassShouldContainConcreteMethods`, or a shorter version of it